rclcpp/include/rclcpp/executor.hpp

```diff
@@ -107,11 +109,13 @@ class Executor
     }
   }
 
-  void spin_node_once(rclcpp::node::Node::SharedPtr & node, bool nonblocking = false)
+  template<typename Rep = int64_t, typename Period = std::milli>
+  void spin_node_once(rclcpp::node::Node::SharedPtr & node, std::chrono::duration<Rep,
+    Period> timeout = std::chrono::duration<Rep, Period>::zero())
```

Comment:
I believe that makes sense as it follows the convention (sort of) for select, where positive is timed blocking, both sec and nsec zero is non-blocking (special case of timed blocking), and passing NULL for timeout is block forever.

> also, I believe previously there was nothing that blocked indefinitely, by default anything that blocked would block for 1 second

Yeah, but as I see it that was an implementation detail. From `rclcpp`'s perspective it did block indefinitely.